classes/db/DbMySQLi.php

```diff
@@ -448,27 +448,27 @@ public static function checkSelectPrivilege($server, $user, $pwd, $db, $prefix,
             return false;
         }
 
-        if ($engine === null) {
-            $engine = 'MyISAM';
+        $enginesToTest = ['InnoDB', 'MyISAM'];
+        if ($engine !== null) {
+            $enginesToTest = [$engine];
         }
 
-        // Create a table
-        $link->query('
-		CREATE TABLE `' . $prefix . 'test` (
-			`test` tinyint(1) unsigned NOT NULL
-		) ENGINE=' . $engine);
+        foreach ($enginesToTest as $engineToTest) {
+            $link->query('
```

Comment:
I'm sorry I wasn't very clear, you should test all result when using $link->query, not only the select request.

Reply:
ok, but it makes me think if we really should waste any more time on this. That is quite ugly low level code and I don't get it why in 2021 any kind of MySQL based application should still support MyISAM. We're talking about the installer here, and you guys could easily remove MyISAM support for the installer, without breaking any existing installations that are still stuck on MyISAM tables.
Are you aware how bad MyISAM performs in MySQL 8.0?

Reply:
I would leave it as is, because the method is called `checkSelectPrivilege()`, only checking `SELECT` priv. If `CREATE` fails, the `SELECT` would also fail, and if `CREATE` and `SELECT` works, but `DROP` fails, the user is left with a `test` table lying around which will not do any harm.
You really should try to support any kind of crippled permission setups, but rather clearly document how to set up a MySQL user for PrestaShop. Usually installers don't test every single permission and this should be the user's responsibility to set it up correctly.

Reply:
> Are you aware how bad MyISAM performs in MySQL 8.0?

Yes, but not everyone is running it on MySQL.
And not everyone has a new server for all instances. Unfortunately, MyISAM is still supported :( 

> I would leave it as is, because the method is called checkSelectPrivilege(), only checking SELECT priv

You're completely right :p